Sources/ClientRuntime/Networking/Http/Middlewares/RequestBody/BodyMiddleware.swift

```diff
@@ -6,22 +6,22 @@
 //
 
 import struct Foundation.Data
-import typealias SmithyReadWrite.DocumentWritingClosure
+import protocol SmithyReadWrite.SmithyWriter
 import typealias SmithyReadWrite.WritingClosure
 
 public struct BodyMiddleware<OperationStackInput,
                              OperationStackOutput,
-                             Writer>: Middleware {
+                             Writer: SmithyWriter>: Middleware {
     public let id: Swift.String = "BodyMiddleware"
 
-    let documentWritingClosure: DocumentWritingClosure<OperationStackInput, Writer>
+    let rootNodeInfo: Writer.NodeInfo
```

Comment:
The document writing closure used to carry in the root node info.  Since document writing closures are eliminated, the root node info is passed in directly.